e2e_playwright/st_fragment_multiple_fragments.py

```diff
@@ -24,5 +24,13 @@ def my_fragment(n):
         st.write(f"uuid in fragment {n}: {uuid4()}")
 
 
+@st.fragment
+def exception_raising_fragment():
+    if st.checkbox("Raise Exception", value=True):
+        raise RuntimeError("This is an exception raised in a fragment")
+
+
 my_fragment(1)
+# fragment that raises an exception does not stop the flow
+exception_raising_fragment()
```

Comment:
Hm, this behavior is something that I think is worth getting product team input on.

By the time we're rerunning fragments, we definitely want fragments that raise exceptions to not interrupt other fragments as we want to think of any fragment run as independent. In the context of a full script run, though, my intuition is that we'll want exceptions in a fragment to stop script execution like any other exception.

Reply:
@jrieke @sfc-gh-jcarroll What do you think? Right now, fragment exceptions are shown to the user but do not block the full app run right now. Do you want this to change and stop the app execution when a fragment raises an error during full-app execution?
My opinion is that it should be fine to continue app execution which might still succeed; and if the fragment run was required I expect the full app code to error out later anyways. Since we have somewhat control over the fragment code (in contrast to the full app code), my thinking is that we can handle these errors more gracefully. But it sounds Vincent's intuition is different, so what do you think?

Reply:
I think it should halt the full script run execution, 100%

Reply:
Alright, we have decided to stop app execution in case an exception is raised by a fragment during a full app run